tests/config/basic/composites-bad.yaml

```diff
@@ -1,11 +1,10 @@
 composites:
-  storageClasses:
+  names:
```

Comment:
What do you think about just dropping this `names` level and letting "default" be a specially-recognized key at the next level?  It just feels like it's a superfluous node in the vast majority of config trees, which will inherit the default from somewhere.  And anyone who wants to name their DatasetType "default" deserves what they get :slightly_smiling_face: .

Reply:
I did think about that, but then I worried that it would make it very difficult to add any other configuration items to that config in the future.

Reply:
How about changing `names` to `disassembly` (then at least the True/False relates to the name)?

Reply:
`disassembled`?